### PR TITLE
Reload on renew, update file layout, enable bionic

### PR DESCRIPTION
**Pull request by chris-sanders**
Updated layout to match latest template
Added a reload for haproxy on cert renew
Set Bionic as stable because it is passing all tests, after this lands we can remove bionic form canary.

### CODE REVIEW

**Comment by chris-sanders** [comment]
Part of this will conflict with #11 b/c we both decided to update the layout :) 
I'm happy to rebase this on top of 11, however if we decide to rework the way 11 is implemented a bit and this is ready to go I would like to get this service reload implemented. I think this is the cause of expired certificates.

File: metadata.yaml [hunk 1]
```diff
@@ -10,6 +10,7 @@ tags:
 subordinate: false
 series:
  - xenial
+ - bionic
```
Comment (digitalrane) [inline metadata.yaml]: In my merge, I had left bionic out, and instead marked it as a canary for functional testing. Perhaps we should do that for a while before marking it supported?

File: tests/functional/test_deploy.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import time
+
+# from juju.model import Model
+
+# Treat tests as coroutines
+pytestmark = pytest.mark.asyncio
+
+juju_repository = os.getenv('JUJU_REPOSITORY', '.').rstrip('/')
+series = ['xenial',
+          pytest.param('bionic', marks=pytest.mark.xfail(reason='canary')),
```
Comment (digitalrane) [inline tests/functional/test_deploy.py:14]: Per comment above we should probably remove bionic from metadata until we've run canary functional tests for a while

File: tests/functional/test_deploy.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+async def test_wrong_login(app):
+    unit = app.units[0]
+    page = requests.get('http://{}:{}/{}'.format(unit.public_address, 9000, 'ha-stats'),
+                        auth=requests.auth.HTTPBasicAuth('admin', 'fail')
+                        )
+    assert page.status_code == 401
+
+
+@pytest.mark.usefixtures("nostats")
```
Comment (digitalrane) [inline tests/functional/test_deploy.py:104]: Oh, I just now totally understand what this is doing. Very clever. I'll likely just cancel my other MP and rework it into something more generic anyhow, so I think we should go with these tests.

**Review by digitalrane — CHANGES_REQUESTED** [review]
The only thing I'd like is for bionic to be removed from metadata until we've tested it for a bit, otherwise, LGTM

File: metadata.yaml (hunk 1 above)
Comment (chris-sanders) [reply metadata.yaml]: ./agree I'll remove it

**Review by digitalrane — APPROVED** [review]
LGTM

**Pull request merged**